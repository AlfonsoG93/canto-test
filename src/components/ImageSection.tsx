import testImage from "../images/testimage.jpeg";
import SvgContainer from "./SvgContainer";

interface ImageSectionProps {
	approved: boolean;
	rejected: boolean;
}


export default function ImageSection(props: ImageSectionProps) {
	return (
		<div
			
			data-testid={"image-container"}
			style={{paddingBottom: "2rem"}}
		>
			<div
				
				data-testid={"image"}
				style={{
				backgroundImage: `url(${testImage})`, backgroundRepeat: "no-repeat",
				backgroundSize: "contain",
				height: "70vh",
				padding: "1rem",
				margin: "2rem",
				justifyContent: "flex-end",
				display: "flex",
				
			}}>
				<SvgContainer approved={props.approved} rejected={props.rejected}/>
			</div>
		</div>
	);
	
}