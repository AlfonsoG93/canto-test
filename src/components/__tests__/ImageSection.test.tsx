import React from "react";
import {cleanup, render, screen} from "@testing-library/react";
import ImageSection from "../ImageSection";
import renderer from "react-test-renderer";

describe("Test ImageSection component", () => {
	afterEach(() => {
		cleanup();
	});
	
	test("Should render component", () => {
		render(<ImageSection approved={false} rejected={false} />);
		const container = screen.getByTestId("image-container");
		const image = screen.getByTestId("image");
		
		expect(container).toBeInTheDocument();
		expect(image).toBeInTheDocument();
		
	});
	
	test("matches snapshot", () => {
		const tree = renderer.create(<ImageSection approved={false} rejected={false}/>).toJSON();
		expect(tree).toMatchSnapshot();
		
	});
});