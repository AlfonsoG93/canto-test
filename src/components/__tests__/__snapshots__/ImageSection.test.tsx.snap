// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component matches snapshot 1`] = `
<div
  data-testid="image-container"
  style={
    Object {
      "paddingBottom": "2rem",
    }
  }
>
  <div
    data-testid="image"
    style={
      Object {
        "backgroundImage": "url(testimage.jpeg)",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "contain",
        "display": "flex",
        "height": "70vh",
        "justifyContent": "flex-end",
        "margin": "2rem",
        "padding": "1rem",
      }
    }
  >
    <div
      data-testid="result-container"
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": "5px",
          "display": "flex",
          "maxHeight": "12%",
        }
      }
    />
  </div>
</div>
`;

exports[`Test ImageSection component matches snapshot 1`] = `
<div
  data-testid="image-container"
  style={
    Object {
      "paddingBottom": "2rem",
    }
  }
>
  <div
    data-testid="image"
    style={
      Object {
        "backgroundImage": "url(testimage.jpeg)",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "contain",
        "display": "flex",
        "height": "70vh",
        "justifyContent": "flex-end",
        "margin": "2rem",
        "padding": "1rem",
      }
    }
  >
    <div
      data-testid="result-container"
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": "5px",
          "display": "flex",
          "maxHeight": "12%",
        }
      }
    />
  </div>
</div>
`;
